packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/project/repositories/ProjectSearchRepositoryDelegate.java

```diff
@@ -92,31 +95,46 @@ public Optional<Project> findById(String projectId) {
     }
 
     @Override
-    public List<Project> findAllBefore(String cursorProjectId, int limit) {
+    public List<Project> findAllBefore(String cursorProjectId, int limit, Map<String, Object> filter) {
         List<Project> projectsBefore = null;
         if (limit > 0) {
-            var projects = this.getAllProjectsQuery(FIND_ALL_BEFORE, cursorProjectId, limit + 1);
+            var projects = this.getAllProjectsQuery(FIND_ALL_BEFORE, cursorProjectId, limit + 1, filter);
             projectsBefore = projects.subList(0, Math.min(projects.size(), limit));
         }
         return projectsBefore;
     }
 
     @Override
-    public List<Project> findAllAfter(String cursorProjectId, int limit) {
+    public List<Project> findAllAfter(String cursorProjectId, int limit, Map<String, Object> filter) {
         List<Project> projectsAfter = null;
         if (limit > 0) {
-            var projects = this.getAllProjectsQuery(FIND_ALL_AFTER, cursorProjectId, limit + 1);
+            var projects = this.getAllProjectsQuery(FIND_ALL_AFTER, cursorProjectId, limit + 1, filter);
             projectsAfter = projects.subList(0, Math.min(projects.size(), limit));
         }
         return projectsAfter;
     }
 
-    private List<Project> getAllProjectsQuery(String sqlQuery, String cursorProjectId, int limit) {
-        return this.jdbcClient
-                .sql(sqlQuery)
+    private List<Project> getAllProjectsQuery(String sqlQuery, String cursorProjectId, int limit, Map<String, Object> filter) {
+        var statement = this.jdbcClient.sql(sqlQuery)
                 .param(CURSOR_PROJECT_ID, cursorProjectId)
-                .param(LIMIT, limit)
+                .param(LIMIT, limit);
+
+        this.handleFilter(statement, filter, "name", "equals");
```

Comment:
Don't modify the statement, extract the relevant parameters beforehand

Reply:
Done